### PR TITLE
Add utility functions to handle json <-> pb conversion

### PR DESCRIPTION
**Pull request by NonStatic2014**

### CODE REVIEW

File: cmake/onnxruntime_hosting.cmake [hunk 1]
```diff
@@ -27,13 +27,43 @@ find_package(Boost 1.69 COMPONENTS system context thread program_options REQUIRE
 set(re2_src ${REPO_ROOT}/cmake/external/re2)
 
 # Setup source code
+file(GLOB_RECURSE onnxruntime_hosting_lib_srcs
+        "${ONNXRUNTIME_ROOT}/hosting/util.cc"
+        )
 file(GLOB_RECURSE onnxruntime_hosting_srcs
-    "${ONNXRUNTIME_ROOT}/hosting/*.h"
-    "${ONNXRUNTIME_ROOT}/hosting/*.cc"
+    "${ONNXRUNTIME_ROOT}/hosting/main.cc"
```
Comment (tmccrmck) [inline cmake/onnxruntime_hosting.cmake]: Could you format this whole file consistently?

File: onnxruntime/hosting/util.cc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <google/protobuf/util/json_util.h>
+#include "util.h"
+
+namespace protobufutil = google::protobuf::util;
+
+namespace onnx {
```
Comment (tmccrmck) [inline onnxruntime/hosting/util.cc]: `onnxruntime`

File: onnxruntime/hosting/util.cc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  protobufutil::JsonPrintOptions options;
+  options.add_whitespace = false;
+  options.always_print_primitive_fields = false;
+  options.always_print_enums_as_ints = false;
+  options.preserve_proto_field_names = false;
+
+  protobufutil::Status result = MessageToJsonString(response, &json_string, options);
+  return result;
+}
+}
```
Comment (tmccrmck) [inline onnxruntime/hosting/util.cc]: // namespace <foo>

File: onnxruntime/hosting/util.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
```
Comment (tmccrmck) [inline onnxruntime/hosting/util.h:3]: Match other header guards

File: cmake/onnxruntime_hosting.cmake [hunk 5]
```diff
@@ -27,13 +27,43 @@ find_package(Boost 1.69 COMPONENTS system context thread program_options REQUIRE
 set(re2_src ${REPO_ROOT}/cmake/external/re2)
 
 # Setup source code
+file(GLOB_RECURSE onnxruntime_hosting_lib_srcs
+        "${ONNXRUNTIME_ROOT}/hosting/util.cc"
```
Comment (tmccrmck) [inline cmake/onnxruntime_hosting.cmake]: If we are going to make a separate lib, should we move it to it's own folder?

File: cmake/onnxruntime_hosting.cmake (hunk 1 above)
Comment (NonStatic2014) [reply cmake/onnxruntime_hosting.cmake]: Done

File: cmake/onnxruntime_hosting.cmake (hunk 5 above)
Comment (NonStatic2014) [reply cmake/onnxruntime_hosting.cmake]: I will merge it with http utilities in the next PR.

**Review by tmccrmck — APPROVED** [review]

**Pull request merged**